r/tests/testthat/test-dplyr.R

```diff
@@ -524,7 +524,7 @@ test_that("is.finite(), is.infinite(), is.nan()", {
       ) %>% collect(),
     df
   )
-  skip("is.nan() evaluates to NA on NA values (ARROW-12850)")
+  # is.nan() evaluates to FALSE on NA_real_ (ARROW-12850)
```

Comment:
Should this comment be updated to note that we've worked around this in R but that perhaps it should be handled in C++ by ARROW-12850?

Reply:
I created ARROW-13366 for this. I'll add `TODO` comments that reference this in the `is.nan` implementations, not here in the tests.

Reply:
I also created ARROW-13367 for handling the `is.na` treatment of `NaN`s with a C++ option. `TODO` comments for these are added in 6c2876335b5b9481f8ac9edd069e1e130e486bac